Client/Frontend/Widgets/SnackBar.swift

```diff
@@ -0,0 +1,225 @@
+        self.title = title
+        self.callback = callback
+    }
+
+    func onClick() {
+        println("onClick")
+        callback(bar: bar)
+    }
+
+}
+
+/* Presents some information to the user. Can optionally include some buttons and
+ * an image. Usage:
+ * let bar = SnackBar(text: "This is some text in the snackbar.",
+ *     img: UIImage(named: "bookmark"),
+ *     buttons: [
+ *         SnackButton(title: "OK", { _ in println("OK") }),
+ *         SnackButton(title: "Cancel", { _ in println("Cancel") }),
+ *         SnackButton(title: "Maybe", { _ in println("Maybe") })]
+ *     )
+ *     view.addSubview(bar)
+ *     bar.snp_makeConstraints({ make in
+ *         make.bottom.equalTo(view)
+ *         make.left.width.equalTo(self.view)
+ *     })
+ */
+class SnackBar: UIView {
+    let imageView: UIImageView
+    let textView: UITextView
+    let buttonsView: Toolbar
+    let font = UIFont(name: "FiraSans-Regular", size: 13)
+    private var targets = [SnackButton]()
+
+    convenience init(text: String, img: UIImage?, buttons: [SnackButton]?) {
+        let font = UIFont(name: "FiraSans-Regular", size: 13)
+        var attributes = [NSObject: AnyObject]()
+        attributes[NSFontAttributeName] = font;
+        attributes[NSBackgroundColorAttributeName] = UIColor.clearColor()
+        let attrText = NSAttributedString(string: text, attributes: attributes)
+        self.init(attrText: attrText, img: img, buttons: buttons)
+    }
+
+    init(attrText: NSAttributedString, img: UIImage?, buttons: [SnackButton]?) {
+        imageView = UIImageView()
+        textView = UITextView()
+        buttonsView = Toolbar()
+        super.init()
+        imageView.image = img
+        textView.attributedText = attrText
+        if let buttons = buttons {
+            for button in buttons {
+                addButton(button)
+            }
+        }
+        setup()
+    }
+
+    private override init(frame: CGRect) {
+        imageView = UIImageView()
+        textView = UITextView()
+        buttonsView = Toolbar()
+
+        super.init(frame: frame)
+    }
+
+    private func setup() {
+        textView.backgroundColor = nil
+
+        let effect = UIVisualEffectView(effect: UIBlurEffect(style: UIBlurEffectStyle.ExtraLight))
+        addSubview(effect)
+
+        addSubview(imageView)
+        addSubview(textView)
+        addSubview(buttonsView)
+        effect.snp_remakeConstraints { make in
+            make.edges.equalTo(self)
+            return
+        }
+
+        self.backgroundColor = UIColor.clearColor()
+        buttonsView.drawTopBorder = true
+        buttonsView.drawBottomBorder = true
+        buttonsView.drawSeperators = true
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // By default, snackbars persist forever
+    func shouldPersist(browser: Browser) -> Bool {
+        return true
+    }
+
+    override func layoutSubviews() {
```

Comment:
You should not setup constraints in `layoutSubviews` but override `updateConstraints` instead.

> Custom views that set up constraints themselves should do so by overriding this method. When your custom view notes that a change has been made to the view that invalidates one of its constraints, it should immediately remove that constraint, and then call setNeedsUpdateConstraints to note that constraints need to be updated.